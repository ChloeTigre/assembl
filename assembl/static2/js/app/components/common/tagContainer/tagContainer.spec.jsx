// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.3';
/* eslint-enable */

import TagContainer from './tagContainer';

configure({ adapter: new Adapter() });

describe('<TagOnPost /> - with shallow', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TagContainer />);
  });

  it('should render a title', () => {
    expect(wrapper.find('div [className="title"]')).toHaveLength(1);
  });

  it('should render a list of tags', () => {
    expect(wrapper.find('div [className="tag-list"]')).toHaveLength(1);
  });
});