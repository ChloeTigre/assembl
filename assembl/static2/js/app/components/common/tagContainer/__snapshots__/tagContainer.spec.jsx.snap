// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tag On Post|TagContainer admin 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagContainer
    isAdmin={true}
    onTagListUpdateCallback={[Function]}
    postId="0"
    tagList={
      Array [
        Object {
          "id": "0",
          "text": "Habitat et SDF",
        },
        Object {
          "id": "1",
          "text": "Facilitation",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          Ajoutez des mots-clés relatifs à ce post :
        </div>,
        <div
          className="tag-list"
        >
          <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={true}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "tag-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ajoutez des mots-clés relatifs à ce post :",
          "className": "title",
        },
        "ref": null,
        "rendered": "Ajoutez des mots-clés relatifs à ce post :",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={true}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />,
          "className": "tag-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alreadyAddedWarningMessage": "Déja ajouté",
            "isAdmin": true,
            "onTagListUpdateCallback": [Function],
            "postId": "0",
            "tagsList": Array [
              Object {
                "id": "0",
                "text": "Habitat et SDF",
              },
              Object {
                "id": "1",
                "text": "Facilitation",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            Ajoutez des mots-clés relatifs à ce post :
          </div>,
          <div
            className="tag-list"
          >
            <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={true}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "tag-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ajoutez des mots-clés relatifs à ce post :",
            "className": "title",
          },
          "ref": null,
          "rendered": "Ajoutez des mots-clés relatifs à ce post :",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={true}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />,
            "className": "tag-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alreadyAddedWarningMessage": "Déja ajouté",
              "isAdmin": true,
              "onTagListUpdateCallback": [Function],
              "postId": "0",
              "tagsList": Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Storyshots Tag On Post|TagContainer default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagContainer
    isAdmin={false}
    onTagListUpdateCallback={[Function]}
    postId="0"
    tagList={
      Array [
        Object {
          "id": "0",
          "text": "Habitat et SDF",
        },
        Object {
          "id": "1",
          "text": "Facilitation",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          Mots-clés relatifs à ce post :
        </div>,
        <div
          className="tag-list"
        >
          <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={false}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "tag-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mots-clés relatifs à ce post :",
          "className": "title",
        },
        "ref": null,
        "rendered": "Mots-clés relatifs à ce post :",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={false}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />,
          "className": "tag-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alreadyAddedWarningMessage": "Déja ajouté",
            "isAdmin": false,
            "onTagListUpdateCallback": [Function],
            "postId": "0",
            "tagsList": Array [
              Object {
                "id": "0",
                "text": "Habitat et SDF",
              },
              Object {
                "id": "1",
                "text": "Facilitation",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            Mots-clés relatifs à ce post :
          </div>,
          <div
            className="tag-list"
          >
            <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={false}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "tag-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mots-clés relatifs à ce post :",
            "className": "title",
          },
          "ref": null,
          "rendered": "Mots-clés relatifs à ce post :",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={false}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />,
            "className": "tag-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alreadyAddedWarningMessage": "Déja ajouté",
              "isAdmin": false,
              "onTagListUpdateCallback": [Function],
              "postId": "0",
              "tagsList": Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Storyshots Tag On Post|TagContainer playground 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagContainer
    isAdmin={false}
    onTagListUpdateCallback={[Function]}
    postId="0"
    tagList={
      Array [
        Object {
          "id": "0",
          "text": "Habitat et SDF",
        },
        Object {
          "id": "1",
          "text": "Facilitation",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          Mots-clés relatifs à ce post :
        </div>,
        <div
          className="tag-list"
        >
          <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={false}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "tag-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mots-clés relatifs à ce post :",
          "className": "title",
        },
        "ref": null,
        "rendered": "Mots-clés relatifs à ce post :",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Apollo(Apollo(Component)))
            alreadyAddedWarningMessage="Déja ajouté"
            isAdmin={false}
            onTagListUpdateCallback={[Function]}
            postId="0"
            tagsList={
              Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ]
            }
          />,
          "className": "tag-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alreadyAddedWarningMessage": "Déja ajouté",
            "isAdmin": false,
            "onTagListUpdateCallback": [Function],
            "postId": "0",
            "tagsList": Array [
              Object {
                "id": "0",
                "text": "Habitat et SDF",
              },
              Object {
                "id": "1",
                "text": "Facilitation",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            Mots-clés relatifs à ce post :
          </div>,
          <div
            className="tag-list"
          >
            <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={false}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "tag-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mots-clés relatifs à ce post :",
            "className": "title",
          },
          "ref": null,
          "rendered": "Mots-clés relatifs à ce post :",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Apollo(Apollo(Component)))
              alreadyAddedWarningMessage="Déja ajouté"
              isAdmin={false}
              onTagListUpdateCallback={[Function]}
              postId="0"
              tagsList={
                Array [
                  Object {
                    "id": "0",
                    "text": "Habitat et SDF",
                  },
                  Object {
                    "id": "1",
                    "text": "Facilitation",
                  },
                ]
              }
            />,
            "className": "tag-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alreadyAddedWarningMessage": "Déja ajouté",
              "isAdmin": false,
              "onTagListUpdateCallback": [Function],
              "postId": "0",
              "tagsList": Array [
                Object {
                  "id": "0",
                  "text": "Habitat et SDF",
                },
                Object {
                  "id": "1",
                  "text": "Facilitation",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
